### PR TITLE
[ADD] Add new module website_sale_partner_type

### PR DESCRIPTION
**Pull request by hurrinico**
Add Partner Type selection on Checkout view
Actually is WIP

### CODE REVIEW

**Comment by eLBati** [comment]
Thanks @hurrinico 

Please see travis

```
======== Testing test_flake8 ========

website_sale_partner_type/controllers/main.py:6:1: F401 'http' imported but unused

website_sale_partner_type/controllers/main.py:14:9: F841 local variable 'cr' is assigned to but never used

website_sale_partner_type/controllers/main.py:14:13: F841 local variable 'uid' is assigned to but never used

website_sale_partner_type/controllers/main.py:14:18: F841 local variable 'context' is assigned to but never used

website_sale_partner_type/controllers/main.py:14:27: F841 local variable 'registry' is assigned to but never used

website_sale_partner_type/controllers/main.py:16:9: F841 local variable 'res' is assigned to but never used

website_sale_partner_type/controllers/main.py:35:13: F841 local variable 'uid' is assigned to but never used

website_sale_partner_type/controllers/main.py:35:80: E501 line too long (95 > 79 characters)

======== Testing test_pylint ========

************* Module website_sale_partner_type

website_sale_partner_type/__init__.py:5: [W0403(relative-import), ] Relative import 'controllers', should be 'website_sale_partner_type.controllers'

************* Module website_sale_partner_type.__openerp__

website_sale_partner_type/__openerp__.py:5: [W0109(duplicate-key), ] Duplicate key 'author' in dictionary
```

**Comment by hurrinico** [comment]
@eLBati travis is green now

**Comment by rafaelbn** [comment]
Hi @hurrinico please check runbot errors. Thanks!!

**Comment by rafaelbn** [comment]
Hi @hurrinico please check runbot errors. Thanks!!

´´´
2016-07-05 10:52:52 	CRITICAL 	server 	openerp.service.server:949 preload_registries

Failed to initialize database `3150072-108-1d6283-base`.
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/service/server.py", line 936, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/modules/registry.py", line 370, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/modules/loading.py", line 311, in load_modules
    modobj.update_list(cr, SUPERUSER_ID)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/addons/base/module/module.py", line 651, in update_list
    handler.load_addons()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/http.py", line 1316, in load_addons
    m = __import__('openerp.addons.' + module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/modules/module.py", line 80, in load_module
    mod = imp.load_module('openerp.addons.' + module_part, f, path, descr)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/addons/product_multi_image/__init__.py", line 5, in <module>
    from .hooks import pre_init_hook
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/addons/product_multi_image/hooks.py", line 5, in <module>
    from openerp.addons.base_multi_image.hooks import pre_init_hook_for_submodules
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/modules/module.py", line 79, in load_module
    f, path, descr = imp.find_module(module_part, ad_paths)
ImportError: No module named base_multi_image
´´´

**Comment by rafaelbn** [comment]
Hi @hurrinico please check runbot errors. Thanks!!

```
2016-07-05 10:52:52 CRITICAL server openerp.service.server:949 preload_registries

Failed to initialize database 3150072-108-1d6283-base.
Traceback (most recent call last):
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/service/server.py", line 936, in preload_registries
registry = RegistryManager.new(dbname, update_module=update_module)
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/modules/registry.py", line 370, in new
openerp.modules.load_modules(registry.db, force_demo, status, update_module)
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/modules/loading.py", line 311, in load_modules
modobj.update_list(cr, SUPERUSER_ID)
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/api.py", line 268, in wrapper
return old_api(self, args, *kwargs)
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/addons/base/module/module.py", line 651, in update_list
handler.load_addons()
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/http.py", line 1316, in load_addons
m = __import_('openerp.addons.' + module)
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/modules/module.py", line 80, in load_module
mod = imp.load_module('openerp.addons.' + module_part, f, path, descr)
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/addons/product_multi_image/init.py", line 5, in
from .hooks import pre_init_hook
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/addons/product_multi_image/hooks.py", line 5, in
from openerp.addons.base_multi_image.hooks import pre_init_hook_for_submodules
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150072-108-1d6283/openerp/modules/module.py", line 79, in load_module
f, path, descr = imp.find_module(module_part, ad_paths)
ImportError: No module named base_multi_image
```

**Comment by atchuthan** [comment]
@hurrinico this issue could be resolved by adding **server-tools** in oca_dependencies.txt
https://github.com/OCA/e-commerce/blob/8.0/oca_dependencies.txt

**Comment by hurrinico** [comment]
done @atchuthan @rafaelbn

**Comment by eLBati** [comment]
@atchuthan @hurrinico it seems it did not solve

**Comment by eLBati** [comment]
@OCA/e-commerce-maintainers  do you know why?

**Comment by atchuthan** [comment]
@eLBati this issue might be solved with this commit https://github.com/OCA/product-attribute/commit/c26675cf89146e00c7a707a1d41912ea105efc46.  Could restart runbot build?

@hurrinico sorry about the comment of oca-dependencies.txt, it seems there was an issue with an import statement. Please remove the server-tools in it.

**Comment by atchuthan** [comment]
@eLBati @hurrinico after restarting runbot build manually, it is a success now.

**Comment by paolovalier** [comment]
👍

**Comment by archetipo** [comment]
:+1:

**Comment by atchuthan** [comment]
:+1:

File: oca_dependencies.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@ bank-statement-reconcile
 product-attribute
 sale-workflow
 website
+server-tools
```
Comment (atchuthan) [inline oca_dependencies.txt]: @hurrinico please remove server-tools from oca_dependencies.txt

**Comment by hurrinico** [comment]
Done @atchuthan

**Comment by rafaelbn** [comment]
@hurrinico here https://github.com/OCA/e-commerce/pull/108#issuecomment-230460218 travis was green and now fails

**Comment by hurrinico** [comment]
@rafaelbn quite strange error

**Comment by rafaelbn** [comment]
This happens after your fix https://github.com/OCA/e-commerce/pull/108/commits/c9e01aedf9168aeb5925340b0103c9a11571af78

**Comment by rafaelbn** [comment]
We cannot merge without fixing travis. Take a llok

```
  Found existing installation: paramiko 1.7.7.1

    Not uninstalling paramiko at /usr/lib/python2.7/dist-packages, outside environment /home/travis/virtualenv/python2.7_with_system_site_packages

  Can't roll back paramiko; was not uninstalled

Traceback (most recent call last):

  File "/usr/lib/python2.7/logging/__init__.py", line 870, in emit

  File "/usr/lib/python2.7/logging/__init__.py", line 832, in flush

IOError: [Errno 5] Input/output error

Logged from file req_uninstall.py, line 139

  Exception:

  Traceback (most recent call last):

    File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pip/basecommand.py", line 232, in main

    File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pip/commands/install.py", line 382, in run

    File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pip/utils/build.py", line 38, in __exit__

    File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pip/utils/build.py", line 42, in cleanup

    File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 58, in rmtree

    File "/usr/lib/python2.7/shutil.py", line 237, in rmtree

    File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 66, in rmtree_errorhandler

  OSError: [Errno 5] Input/output error: '/tmp/pip-build-BQsOg3'

Traceback (most recent call last):

  File "/usr/lib/python2.7/logging/__init__.py", line 870, in emit

  File "/usr/lib/python2.7/logging/__init__.py", line 832, in flush

IOError: [Errno 5] Input/output error

Logged from file basecommand.py, line 258

Traceback (most recent call last):

  File "/home/travis/maintainer-quality-tools/travis/clone_oca_dependencies", line 119, in <module>

  File "/home/travis/maintainer-quality-tools/travis/clone_oca_dependencies", line 104, in run

  File "/usr/lib/python2.7/subprocess.py", line 511, in check_call

subprocess.CalledProcessError: Command '['pip', 'install', '-r', '/home/travis/dependencies/server-tools/requirements.txt']' returned non-zero exit status 2

/home/travis/build.sh: line 77: uname: command not found

/home/travis/build.sh: line 85: date: command not found

The command "travis_install_nightly" failed and exited with 1 during .

Your build has been stopped.

/home/travis/build.sh: line 109: /home/travis/.bash_logout: Input/output error

/home/travis/build.sh: line 109: /etc/bash.bash_logout: Input/output error
```

File: website_sale_partner_type/README.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=========================
+Website Sale Partner Type
+=========================
+
+This module adds partner type on checkout and set on cutomer and
```
Comment (rafaelbn) [inline website_sale_partner_type/README.rst]: s/cutomer/customer

**Comment by rafaelbn** [comment]
Hi @hurrinico , have you tested in runbot? I'm not able to finish the sale. After confirming the sale I press "Confirm" button and it doesn't continue.

**Comment by eLBati** [comment]
Tried to restart travis build but got

> This job can't be restarted

Maybe somebody with write rights?

**Comment by hurrinico** [comment]
hi @rafaelbn can you please tell me the steps to reproduce the problem?
Local i had a similar problem and was because i didn't fill some required fields

**Comment by rafaelbn** [comment]
Hi @hurrinico , thanks again for this contribution, I will try to help.
- Go to runbot
- Log in as admin
- uninstall module: website_sale_require_login
- Log out as admin
- Go to shop
- Buy and iPad
- Process checkout (without login)
- Check which are mandatory fields and fill only that fields as in the screenshot (bellow)
- Try to confirm order (you cannot - BUG)


- Now
- Uninstall your module website_sale_partner_type
- Make same test (you can)

Is about required fields? Which field is now required? As you see in screenshot I fill all required fields:
- You are deleting field "You name" witch is mandatory. Why? I guess**\* is because the functionality of this module that I cannot test it that (1) In the case you choose Partner Type "Company" you will create a res.partner with Boolean "is a company" set in true so you don't have place to write the name. (2) If the case you choose "individual" then is a normal process
- You are adding a new field "Street2". Why? This must be done in a separate module



Please rewrite README description / usage in order that anyone can understand which is the purpose of this module and how to test it, it's dificult to understand.

Conceptually is not correct to buy onilne (offline is impossible) without a person name. Maybe do you miss field "You name" when selecting partner typo "company"? Companies don't buy alone, they need someone :smile: 

cc @pedrobaeza https://github.com/OCA/e-commerce/pull/108#issuecomment-231760701

**Comment by hurrinico** [comment]
Hi @rafaelbn ,
Thank you for the review.
Your Name field is Company Name field renamed if partner_type = company (as you can see in the code).
About the bug, i'm quite busy at the moment, i think i'll watch it in the next days

**Comment by rafaelbn** [comment]
Hi @hurrinico , IMHO I think you should add a new field for this purpose and maintain the standard one. It's not recommended to delete or overwrite existing fields. Furthermore, if you do so, BUG will be fixed.
Please add test to your core: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#tests

**Comment by hurrinico** [comment]
Hi @rafaelbn thank you.
At the moment i'm quite busy so i think i'll work on it later. 
For my purpose it's good what i've already done.
Meantime if someone want to do a PR on my branch i'll merge

**Pull request closed without merging**